worker/src/Settings.cpp

```diff
@@ -73,13 +74,9 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 	optind = 1; // Set explicitly, otherwise subsequent runs will fail.
 	opterr = 0; // Don't allow getopt to print error messages.
+
 	while ((c = getopt_long_only(argc, argv, "", options, &optionIdx)) != -1)
 	{
-		if (!optarg)
-		{
-			MS_THROW_TYPE_ERROR("unknown configuration parameter: %s", optarg);
-		}
```

Comment:
Why was this deleted? It seems like an important safety check that caller provided something meaningful as an input.

Reply:
Because the new `--disableLiburing` command line argument doesn't have any value so such a check throws if present. I can check that `optargs` exist for all the other arguments but didn't consider it necessary.

Reply:
Well, but other options do require a value. Maybe we have a test that checks that and it causes memory corruption because you suddenly create a value out of null pointer?

Reply:
I've made this change:

https://github.com/versatica/mediasoup/pull/1442/commits/db2252c4edc64a0f4be53f6d9d51622592ac25ba

Is it enough? What do you mean with "you suddenly create a value out of null pointer"? Command line arguments are created by Node and Rust layers in their `Worker` classes. Tests can not trigger wrong arguments passed to the worker.

Just wondering about this: In `worker/utils.rs`:

```rust
pub(super) fn run_worker_with_channels<OE>(
    id: WorkerId,
    thread_initializer: Option<Arc<dyn Fn() + Send + Sync>>,
    args: Vec<String>,
    worker_closed: Arc<AtomicBool>,
    on_exit: OE,
) -> WorkerRunResult
where
    OE: FnOnce(Result<(), ExitError>) + Send + 'static,
{
    let (channel, prepared_channel_read, prepared_channel_write) =
        Channel::new(Arc::clone(&worker_closed));
    let buffer_worker_messages_guard =
        channel.buffer_messages_for(SubscriptionTarget::String(std::process::id().to_string()));

    std::thread::Builder::new()
        .name(format!("mediasoup-worker-{id}"))
        .spawn(move || {
            if let Some(thread_initializer) = thread_initializer {
                thread_initializer();
            }
            let argc = args.len() as c_int;
            let args_cstring = args
                .into_iter()
                .map(|s| -> CString { CString::new(s).unwrap() })
                .collect::<Vec<_>>();
            let argv = args_cstring
                .iter()
                .map(|arg| arg.as_ptr().cast::<c_char>())
                .collect::<Vec<_>>();
            let version = CString::new(env!("CARGO_PKG_VERSION")).unwrap();

            let status_code = unsafe {
                let (channel_read_fn, channel_read_ctx, _channel_write_callback) =
                    prepared_channel_read.deconstruct();
                let (channel_write_fn, channel_write_ctx, _channel_read_callback) =
                    prepared_channel_write.deconstruct();

                mediasoup_sys::mediasoup_worker_run(
                    argc,
                    argv.as_ptr(),
                    version.as_ptr(),
                    0,
                    0,
                    channel_read_fn,
                    channel_read_ctx,
                    channel_write_fn,
                    channel_write_ctx,
                )
            };
```

Here `args` is a command line arguments string, something like:

```
"--logLevel=warn --disableLiburing"
```

Maybe something dangerous when doing this?:

```rust
let args_cstring = args
                .into_iter()
                .map(|s| -> CString { CString::new(s).unwrap() })
                .collect::<Vec<_>>();
            let argv = args_cstring
                .iter()
                .map(|arg| arg.as_ptr().cast::<c_char>())
                .collect::<Vec<_>>();
```

Reply:
Nope, no problem there IMHO. It just splits the string into these strings:
- "--logLevel=warn"
- "--disableLiburing"

It doesn't do anything like assuming/expecting a "=" symbol, so no danger here.

Reply:
I don't see why commit https://github.com/versatica/mediasoup/commit/db2252c4edc64a0f4be53f6d9d51622592ac25ba should fix this problem. It probably won't and, instead of wasting more time on this, I will change the new command line argument and add a value to it. No time to deal with ancient command line args stuff.

Reply:
I know we're not calling it incorrectly right now, but we could. And that would blow up instead of crashing with a nice message. Do not trust input, at least not to the degree that impacts memory safety.

Reply:
Now it's safe and we don't assume anything. Arg values are now mandatory. See latest changes.